File: Medii de Proiectare si Programare/TrackAndField-Java/src/main/java/repository/UserRepository.java
package repository;

import model.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class UserRepository implements IUserRepository {
    private static final Logger logger= LogManager.getLogger();
    private JdbcUtils dbUtils;

    public UserRepository(Properties props) {
        logger.info("Initializing UserRepository with properties: {} ",props);
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public User findOne(String s) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        User user = null;
        try(PreparedStatement preStmt = con.prepareStatement("select * from User where username = (?)")){
            preStmt.setString(1, s);
            try(ResultSet result = preStmt.executeQuery()){
                String username = result.getString("username");
                String password = result.getString("password");
                user = new User(username, password);
            }
        }catch (SQLException ex)
        {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return user;
    }

    @Override
    public Iterable<User> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<User> users = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from User")){
            try(ResultSet result = preStmt.executeQuery()){
                while(result.next()){
                    String username = result.getString("username");
                    String password = result.getString("password");
                    User user = new User(username, password);
                    users.add(user);
                }
            }
        }catch (SQLException ex)
        {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return users;
    }

    @Override
    public User save(User entity) {
        logger.traceEntry("Saving user {}", entity);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("insert into User(username, password) values (?, ?)")){
            preparedStatement.setString(1, entity.getUsername());
            preparedStatement.setString(2, entity.getPassword());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }catch(SQLException ex)
        {
            logger.error(ex);
            System.err.print("Error DB" + ex);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public User findByUsernameAndPassword(String username, String password) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        User user = null;
        try(PreparedStatement preStmt = con.prepareStatement("select * from User where username = (?) and password = (?)")){
            preStmt.setString(1, username);
            preStmt.setString(1, password);
            try(ResultSet result = preStmt.executeQuery()){
                String username1 = result.getString("username");
                String password1 = result.getString("password");
                user = new User(username1, password1);
            }
        }catch (SQLException ex)
        {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return user;
    }
}

File: Medii de Proiectare si Programare/TrackAndField-C#/TrackAndField/repository/UserRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TrackAndField.model;
using System.Data.SQLite;
using TrackAndField.connectionUtils;
using System.Data.SqlClient;
using Microsoft.Data.Sqlite;

namespace TrackAndField.repository
{
    class UserRepository : IUserRepository
    {
        public IEnumerable<User> FindAll()
        {
            IList<User> users = new List<User>();

            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM User";

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        string username = dataReader.GetString(0);
                        string password = dataReader.GetString(1);
                        User user = new User(username, password);
                        users.Add(user);
                    }
                }
            }

            return users;
        }

        public User FindByUsernameAndPassword(string username, string password)
        {
            User user = null;

            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM User WHERE username = @u AND password = @p";

                var usernameParameter = command.CreateParameter();
                usernameParameter.ParameterName = "@u";
                usernameParameter.Value = username;
                command.Parameters.Add(usernameParameter);

                var passwordParameter = command.CreateParameter();
                passwordParameter.ParameterName = "@p";
                passwordParameter.Value = password;
                command.Parameters.Add(passwordParameter);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        string uName = dataReader.GetString(0);
                        string passwd = dataReader.GetString(1);
                        user = new User(username, password);
                    }
                }
            }

            return user;
        }

        public User FindOne(string id)
        {
            User user = null;

            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM User WHERE username = @id";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@id";
                idParameter.Value = id;
                command.Parameters.Add(idParameter);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        string username = dataReader.GetString(0);
                        string password = dataReader.GetString(1);
                        user = new User(username, password);
                    }
                }
            }

            return user;
        }

        public User Save(User entity)
        {
            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "INSERT INTO User(username, password) VALUES (@u, @p)";

                var usernameParameter = command.CreateParameter();
                usernameParameter.ParameterName = "@u";
                usernameParameter.Value = entity.Username;
                command.Parameters.Add(usernameParameter);

                var passwordParameter = command.CreateParameter();
                passwordParameter.ParameterName = "@p";
                passwordParameter.Value = entity.Password;
                command.Parameters.Add(passwordParameter);

                var result = command.ExecuteNonQuery();

                return entity;
            }
        }
    }
}
